### ISSUE TITLE
[BUG] Document `has_field` does not work for non-default field names

### ISSUE THREAD

**Issue by geoffrp** [voxel51/fiftyone#3346]
### System information

-   **OS Platform and Distribution** (e.g., Linux Ubuntu 16.04): Ubuntu 20.04.3 LTS
-   **Python version** (`python --version`): Python 3.10.2
-   **FiftyOne version** (`fiftyone --version`): 0.21.4
-   **FiftyOne installed from** (pip or source): pip

### Commands to reproduce

Running the following in an `ipython` shell should demonstrate the problem with `has_field`. 

```
import fiftyone as fo
# Create an empty detection
det = fo.Detection()
# Add a custom field to the detection
det['asdf'] = 1.0
# Check with has_field
print(f"Has asdf: {det.has_field('asdf')}")
# Check with `in`
print(f"Has asdf (in): {'asdf' in det}")
```

### Describe the problem

In previous versions, custom fields on a `Detection` would work with `has_field`.  Since updating to version 0.21.4, this no longer works for detections regardless of saved status.

### Code to reproduce issue

See above.  Can be demonstrated with the quickstart dataset as well after saving samples

### What areas of FiftyOne does this bug affect?

-   [ ] `App`: FiftyOne application issue
-   [x] `Core`: Core Python library issue
-   [ ] `Server`: FiftyOne server issue

### Willingness to contribute

The FiftyOne Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the FiftyOne codebase?

-   [ ] Yes. I can contribute a fix for this bug independently
-   [x ] Yes. I would be willing to contribute a fix for this bug with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a bug fix at this time

**Comment by brimoor** [comment]
Thanks for catching this, I've patched it in #3349 👍

**Comment by benjaminpkane** [comment]
Resolved by #3349
